MayssaraOmar/assignment3/Word search/src/WordSearchTest.cpp

```diff
@@ -23,114 +22,169 @@ void test_wordSearch::run_all_tests() {
 	uneven_columns_grid();
 	simple_test();
 }
-void test_wordSearch::empty_grid() {
+
+void WordSearchTest::empty_grid() {
+	// setup
+	dictionary dic;
+	setup(&dic);
+	// build grid
 	vector<vector<char> > grid;
+	// build expected set
 	unordered_set<string> expected;
+	// test
 	unordered_set<string> actual;
-	get_words_from_grid(grid, dic, actual);
+	get_words_from_grid(grid, dic, &actual);
 	EXPECT_EQ(expected, actual);
 }
-void test_wordSearch::one_cell_grid_with_match_in_dictionary() {
+
+void WordSearchTest::one_cell_grid_with_match_in_dictionary() {
+	// setup
+	dictionary dic;
+	setup(&dic);
+	// build grid
 	vector<vector<char> > grid(1);
 	grid[0].push_back('m');
+	// build expected set
 	unordered_set<string> expected;
 	expected.insert("m");
+	// test
 	unordered_set<string> actual;
-	get_words_from_grid(grid, dic, actual);
+	get_words_from_grid(grid, dic, &actual);
 	EXPECT_EQ(expected, actual);
 }
-void test_wordSearch::one_cell_grid_with_no_match_in_dictionary() {
+
+void WordSearchTest::one_cell_grid_with_no_match_in_dictionary() {
+	// setup
+	dictionary dic;
+	setup(&dic);
+	// build grid
 	vector<vector<char> > grid(1);
 	grid[0].push_back('z');
+	// build expected set
 	unordered_set<string> expected;
+	// test
 	unordered_set<string> actual;
-	get_words_from_grid(grid, dic, actual);
+	get_words_from_grid(grid, dic, &actual);
 	EXPECT_EQ(expected, actual);
 }
-void test_wordSearch::same_char_grid_with_match_in_dictionary() {
+
+void WordSearchTest::same_char_grid_with_match_in_dictionary() {
+	// setup
+	dictionary dic;
+	setup(&dic);
+	// build grid
 	vector<vector<char> > grid(2);
 	grid[0].push_back('m');
 	grid[0].push_back('m');
 	grid[1].push_back('m');
 	grid[1].push_back('m');
-
+	// build expected set
 	unordered_set<string> expected;
 	expected.insert("m");
+	// test
 	unordered_set<string> actual;
-	get_words_from_grid(grid, dic, actual);
+	get_words_from_grid(grid, dic, &actual);
 	EXPECT_EQ(expected, actual);
 }
-void test_wordSearch::same_char_grid_with_no_match_in_dictionary() {
+
+void WordSearchTest::same_char_grid_with_no_match_in_dictionary() {
+	// setup
+	dictionary dic;
+	setup(&dic);
+	// build grid
 	vector<vector<char> > grid(2);
 	grid[0].push_back('z');
 	grid[0].push_back('z');
 	grid[1].push_back('z');
 	grid[1].push_back('z');
-
+	// build expected set
 	unordered_set<string> expected;
+	// test
 	unordered_set<string> actual;
-	get_words_from_grid(grid, dic, actual);
+	get_words_from_grid(grid, dic, &actual);
 	EXPECT_EQ(expected, actual);
-
 }
-void test_wordSearch::case_sensitive() {
-	vector<vector<char> > grid(3);
 
+void WordSearchTest::case_sensitive() {
+	// setup
+	dictionary dic;
+	setup(&dic);
+	// build grid
+	vector<vector<char> > grid(3);
```

Comment:
This could be done like:
`vector<vector<char> > grid = { {'a', 'a', 'r'}, {'t', 'c', 'D'}} ;`

Reply:
my code doesn't compile with this kind of vector initialization :/

Reply:
My code doesn't compile with this kind of vector initialization :/ I'll leave it as it is for now and I'll look into the problem